### ISSUE TITLE
Value type of TestBed.configureTestingModule method incorrectly given in the documentation

### ISSUE THREAD

**Issue by magoss** [angular/angular#27190]
# 📚 Docs or angular.io bug report

### Description

Documentation says `configureTestingModule()` method does not return value (void), while in fact it returns TestBed instance, thanks to what you are able to chain other static methods. Testing guide encourages doing so, what is contradiction. 

## 🔬 Minimal Reproduction

https://angular.io/api/core/testing/TestBed#configuretestingmodule

### Expected vs Actual Behavior**

Documentation should clearly say that `configureTestingModule()` method returns typeof TestBed.

## 🌍  Your Environment
Angular 7.0.4

**Comment by jenniferfell** [comment]
@petebacondarwin @jbogarthyde : Template issue? dgeni issue? Seems like the doc content in question is generated directly from the method signature. Do we need to add some doc tagging and an explicit return value? Thoughts?

**Comment by petebacondarwin** [comment]
Looks like void return type in the source code... https://github.com/angular/angular/blob/7.1.0/packages/core/testing/src/test_bed.ts#L54-L55

**Comment by ChrisDahmCS** [comment]
@petebacondarwin please forgive me if I am missing a major piece to this, but I wanted to ask more about TestBed vs TestBedStatic.

It seems like importing the [const TestBed](https://github.com/angular/angular/blob/7.1.0/packages/core/testing/src/test_bed.ts#L641) like [this test](https://github.com/angular/angular/blob/7.1.0/packages/common/test/directives/ng_template_outlet_spec.ts#L11) does brings in the [TestBedStatic](https://github.com/angular/angular/blob/7.1.0/packages/core/testing/src/test_bed_common.ts#L51), which does specify the return type as TestBedStatic in the methods.

That being said, [this part of the documentation](https://github.com/angular/angular/blob/7.1.0/packages/core/testing/src/test_bed_common.ts#L51) is a little misleading to me.

Do you think
```
Note: Use TestBed in tests. It will be set to either TestBedViewEngine or TestBedRender3 according to the compiler used.
```
vs something similar to
```
Note: Use TestBed in tests. It will be set to TestBedStatic using either TestBedViewEngine or TestBedRender3 according to the compiler used.
```
would help clear up some of the confusion in the docs?

**Comment by petebacondarwin** [comment]
OK I looked again at this. The problem we have is that TestBed classes have "two" interfaces. The normal "instance" one that they implement (e.g. `TestBed`) and also their "static" methods, which look very similar but are only available on the class, and delegate to the singleton instance of the `TestBed` that has been created when importing `TestBed` as a "constant" (not the interface)!

Unfortunately TypeScript has no concept of a static interface, so the code does not explicitly declare that `TestBedViewEngine`, etc implement this `TestBedStatic` interface, it is more of a convention.

I think that we just need to fix the documentation to spell this out a lot clearer in the docs. cc @jenniferfell

**Comment by ChrisDahmCS** [comment]
@petebacondarwin is it considered to be taboo to have multiple return types on the TestBed interface methods? If the TestBed methods that end up getting the return type of TestBedStatic in the TestBedStatic interface would have `void | TestBedStatic`, then they can more easily point to the TestBedStatic interface naturally in the docs while the TestBedStatic interface still stays specific with it's return types and could slim down on return types it doesn't change.

**Comment by rockument69** [comment]
@kapunahelewong : You may want to pick up this issue if you are working on the testing doc.

**Comment by JeanMeche** [comment]
This has been fixed by #46635
